plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java

```diff
@@ -74,7 +94,14 @@ public IDocument getDocument() {
     }
 
     public SQLDialect getDialect() {
-        return SQLUtils.getDialectFromDataSource(getDataSource());
+        if (dataSourceContainer != null) {
```

Comment:
It must take dialect from DBPDataSource first.
And then from container (in datasource it is more correct because it loads meta info from actual database).